e2e/types/config.go

```diff
@@ -0,0 +1,46 @@
+// SPDX-FileCopyrightText: The RamenDR authors
+// SPDX-License-Identifier: Apache-2.0
+
+package types
+
+import (
+	"maps"
+	"slices"
+)
+
+// Equal return true if config is equal to other config.
+func (c *Config) Equal(o *Config) bool {
+	if o == nil {
+		return false
+	}
+
+	if !c.compareBasicFields(o) {
+		return false
+	}
```

Comment:
This is not helpful, lets inline the checks here. The function will be longer but we have all the trivial code in one place which makes it easier to understand.